pkgs/development/haskell-modules/HACKING.md

```diff
@@ -0,0 +1,229 @@
+- everything will be committed
+
+### Merge `haskell-updates` into `master`
+
+Now it is time to merge the `haskell-updates` PR you opened above.
+
+Before doing this, make sure of the following:
+
+- All Haskell packages that fail to build are correctly marked broken or
+  transitively broken.
+- The `maintained` and `mergeable` jobs are passing on Hydra.
+- The maintainers for any maintained Haskell packages that are newly broken
+  have been pinged on GitHub and given at least a week to fix their packages.
+  This is especially important for widely-used packages like `cachix`.
+
+When you've double-checked these points, go ahead and merge the `haskell-updates` PR.
+After merging, **make sure not to delete the `haskell-updates` branch**, since it
+causes all currently open Haskell-related pull-requests to be automatically closed on GitHub.
+
+### Additional Info
+
+Here are some additional tips that didn't fit in above.
+
+-   It is possible to start a new Hydra evaluation by logging into Hydra with
+    your GitHub or Google account.
+
+-   You should occasionally merge the `master` branch into the
+    `haskell-updates` branch.
+
+    In an ideal world, when we merge `haskell-updates` into `master`, it would
+    cause few Hydra rebuilds on `master`.  Ideally, the `nixos-unstable`
+    channel would never be prevented from progressing because of needing to
+    wait for rebuilding Haskell packages.
+
+    In order to make sure that there are a minimal number of rebuilds after
+    merging `haskell-updates` into `master`, `master` should occasionally be
+    merged into the `haskell-updates` branch.
+
+    This is especially important after `staging` is merged into `master`, since
+    there is a high chance that this will cause all the Haskell packages to
+    rebuild.
+
+    However, as we are working on cleaning up `haskell-updates`, `master` will
+    continually progress.  It may not always be possible to keep the
+    `haskell-updates` branch fully up-to-date with `master` without causing
+    mass-rebuilds on the `haskell-updates` jobset.
+
+-   Make sure never to update the Hackage package hashes in
+    [`pkgs/data/misc/hackage/`](../../../pkgs/data/misc/hackage/), or the
+    pinned Stackage Nightly versions on the release branches (like
+    `release-21.05`).
+
+    This means that the
+    [`update-hackage.sh`](../../../maintainers/scripts/haskell/update-hackage.sh)
+    and
+    [`update-stackage.sh`](../../../maintainers/scripts/haskell/update-stackage.sh)
+    scripts should never be used on the release branches.
+
+    However, changing other files in `./.` and regenerating the package set is encouraged.
+    This can be done with
+    [`regenerate-hackage-packages.sh`](../../../maintainers/scripts/haskell/regenerate-hackage-packages.sh)
+    as described above.
+
+## Contributor Workflow
+
+(TODO: this section is to describe the type of workflow for non-committers to
+contribute to `haskell-updates`)
```

Comment:
I plan on trying to write this at some point in the future.

Possibly either during Zurihac, or the next time it is my turn to take care of the `haskell-updates` workflow.

Reply:
I don't really like the "out of sight, out of mind" quality of the nixpkgs manual.  I think it is way too easy not to update documentation when it doesn't live near the code.

However, I agree that it probably does make the most sense for the contributor workflow to live there.